AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/models/allocate-hearing.model.ts

```diff
@@ -0,0 +1,158 @@
+        return this.hearings.filter(h => h.checked).map(s => s.hearingId);
+    }
+
+    get hasSelectedHearings(): boolean {
+        return this.hearings.some(h => h.checked);
+    }
+
+    get hasPendingChanges(): boolean {
+        const original = this.originalState.map(h => <any>{ id: h.hearing_id, cso: h.allocated_cso }).sort();
+        const current = this.hearings.map(h => <any>{ id: h.hearingId, cso: h.allocatedOfficerUsername }).sort();
+
+        const stringMatch = JSON.stringify(original) === JSON.stringify(current);
+        return !stringMatch;
+    }
+
+    get areAllChecked(): boolean {
+        return this.hearings.length > 0 && this.hearings.every(h => h.checked);
+    }
+
+    assignCsoToSelectedHearings(csoUsername: string, csoId: string): void {
+        this.hearings.forEach(h => {
+            if (h.checked) {
+                h.updateAssignedCso(csoUsername, csoId);
+            }
+        });
+    }
+
+    checkHearing(hearingId: string) {
+        this.hearings.find(x => x.hearingId === hearingId)?.setChecked(true);
+    }
+
+    uncheckHearingAndRevert(hearingId: string) {
+        // reverting to original hearing defaults to unchecked
+        this.revertHearing(hearingId);
+    }
+
+    checkAllHearings(): void {
+        this.hearings.forEach(h => h.setChecked(true));
+    }
+
+    uncheckAllHearingsAndRevert(): void {
+        this.hearings.forEach(h => this.uncheckHearingAndRevert(h.hearingId));
+    }
+
+    revertHearing(hearingId: string): void {
+        const index = this.hearings.findIndex(h => h.hearingId === hearingId);
+        const originalHearing = this.originalState.find(h => h.hearing_id === hearingId);
+        this.hearings[index] = new AllocateHearingItemModel(
+            originalHearing.hearing_id,
+            originalHearing.hearing_date,
+            originalHearing.start_time,
+            originalHearing.duration,
+            originalHearing.case_number,
+            originalHearing.case_type,
+            originalHearing.allocated_cso,
+            false
+        );
+    }
+
+    /**
+     * Update the original state field with the newly updated hearings
+     */
+    updateHearings(updatedHearings: AllocationHearingsResponse[]): void {
+        updatedHearings.forEach(updatedHearing => {
+            const index = this.originalState.findIndex(x => x.hearing_id === updatedHearing.hearing_id);
+            this.originalState[index] = updatedHearing;
+        });
+
+        this.hearings = this.originalState.map(
+            val =>
+                new AllocateHearingItemModel(
+                    val.hearing_id,
+                    val.hearing_date,
+                    val.start_time,
+                    val.duration,
+                    val.case_number,
+                    val.case_type,
+                    val.allocated_cso,
+                    false
+                )
+        );
+    }
+}
+
+export function arraysEqual(a, b): boolean {
```

Comment:
This should maybe be in a helper class, or somewhere shared and generic. Also probaby needs a test for the sonar coverage.

Reply:
that's odd, I noticed sonar wasn't picking this up but I definitely have tests for this and locally the coverage tool picks this up as covered

Reply:
I got rid of it in the end. I forgot I changed the implantation for checking for pending changes because I found the method unreliable